client/src/components/MultiJurisdictionAudit/useCSV.ts

```diff
@@ -0,0 +1,130 @@
+  electionId: string,
+  jurisdictionId: string,
+  csv: File,
+  filePurpose: 'ballot-manifest' | 'batch-tallies'
+): Promise<boolean> => {
+  const formData: FormData = new FormData()
+  formData.append('manifest', csv, csv.name)
+  try {
+    await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/${filePurpose}`,
+      {
+        method: 'PUT',
+        body: formData,
+      }
+    )
+    return true
+  } catch (err) {
+    toast.error(err.message)
+    return false
+  }
+}
+
+const deleteCSVFile = async (
+  electionId: string,
+  jurisdictionId: string,
+  filePurpose: 'ballot-manifest' | 'batch-tallies'
+): Promise<boolean> => {
+  try {
+    return await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/${filePurpose}`,
+      { method: 'DELETE' }
+    )
+  } catch (err) {
+    toast.error(err.message)
+    return false
+  }
+}
+
+const useCSV = (
+  electionId: string,
+  jurisdictionId: string,
+  filePurpose: 'ballot-manifest' | 'batch-tallies'
+): [
+  IFileInfo | null,
+  (csv: File) => Promise<boolean>,
+  () => Promise<boolean>
+] => {
+  const [csv, setCSV] = useState<IFileInfo | null>(null)
+
+  useEffect(() => {
+    ;(async () => {
+      setCSV(await loadCSV(electionId, jurisdictionId, filePurpose))
+    })()
+  }, [electionId, jurisdictionId])
+
+  const uploadCSV = async (csvFile: File): Promise<boolean> => {
+    if (await putCSVFile(electionId, jurisdictionId, csvFile, filePurpose)) {
+      setCSV(await loadCSV(electionId, jurisdictionId, filePurpose))
+      return true
+    }
+    return false
+  }
+
+  const deleteCSV = async (): Promise<boolean> => {
+    if (await deleteCSVFile(electionId, jurisdictionId, filePurpose)) {
+      setCSV(await loadCSV(electionId, jurisdictionId, filePurpose))
+      return true
+    }
+    return false
+  }
+
+  useEffect(() => {
+    const isFinishedProcessing = (fileInfo: IFileInfo) =>
+      !!(fileInfo.processing && fileInfo.processing.completedAt)
+
+    if (!(csv && csv.file) || isFinishedProcessing(csv)) return
+
+    const isComplete = async () => {
+      const fileInfo = await loadCSV(electionId, jurisdictionId, filePurpose)
+      return !!fileInfo && isFinishedProcessing(fileInfo)
+    }
+    const onComplete = async () => {
+      setCSV(await loadCSV(electionId, jurisdictionId, filePurpose))
+    }
+    poll(isComplete, onComplete, err => toast.error(err.message))
+  }, [electionId, jurisdictionId, csv])
+
+  return [csv, uploadCSV, deleteCSV]
+}
+
+export const useBallotManifest = (
+  electionId: string,
+  jurisdictionId: string
+): [
+  IFileInfo | null,
+  (csv: File) => Promise<boolean>,
+  () => Promise<boolean>
+] => useCSV(electionId, jurisdictionId, 'ballot-manifest')
+
+export const useBatchTallies = (
+  electionId: string,
+  jurisdictionId: string
+): [
+  IFileInfo | null,
+  (csv: File) => Promise<boolean>,
+  () => Promise<boolean>
+] => useCSV(electionId, jurisdictionId, 'batch-tallies')
+
+export default useCSV
```

Comment:
this feels like a useful abstraction. it's definitely some complicated logic so good to reuse it. wondering if it's also possible/worthwhile to abstract the CSV upload form component as well (maybe too soon for that)

also noting that I didn't abstract any of this on the backend lol and now I'm like hmm 🤔

Reply:
Yeah I was thinking of abstracting that too, but it looked like it might end up being too much with how many little places need tweaked between the two in order to get the language right. I might still have a go of it though.